beaconclient/multi_beacon_client.go

```diff
@@ -20,6 +20,18 @@ var (
 	ErrBeaconBlock202           = errors.New("beacon block failed validation but was still broadcast (202)")
 )
 
+type BroadcastValidation int
+
+const (
+	Gossip                   BroadcastValidation = iota // lightweight gossip checks only
+	Consensus                                           // full consensus checks, including validation of all signatures and blocks fields
+	ConsensusAndEquivocation                            // the same as `consensus`, with an extra equivocation check
+)
+
+func (b BroadcastValidation) String() string {
+	return [...]string{"gossip", "consensus", "consensus_and_equivocation"}[b]
```

Comment:
Minor nit: technically `BroadcastValidation` is just an `int` and could be any `int` value, which wouldn't work so well here. But I don't think it's worth changing though, as `parseBroadcastValidationString` is safe & you'd have to intentionally make this mistake.

Reply:
I think a simple `if int(b) >= len(broadcastValidationStrings) {` will suffice, will add as part of future PR